crates/miden-tx/src/tests/kernel_tests/test_fpi.rs

```diff
@@ -706,42 +727,48 @@ fn test_simple_nested_fpi() {
             # push the index of desired storage item
             push.0
 
-            # get the hash of the `get_item` account procedure
-            push.{get_item_foreign_hash}
+            # get the hash of the `second_account_foreign_proc` account procedure from the advice stack
+            adv_push.4
 
-            # push the foreign account ID
-            push.{foreign_suffix}.{foreign_prefix}
+            # push the ID of the second foreign account from the advice stack
+            adv_push.2
             # => [foreign_account_id_prefix, foreign_account_id_suffix, FOREIGN_PROC_ROOT, storage_item_index, pad(14)]
```

Comment:
Here we could avoid the advice stack by formatting the second account ID and procedure root directly into the source code.

Reply:
I don't think that it would work: to use the second account ID and proc root in the code of the first account we have to compile the second account first, but procedure in this second account requires the ID and root of the first account. Because of this circular dependency I have to use the advice stack to provide the values which are not available at the moment of account code compilation. I can change it to use hardcoded hashes and IDs just for test purposes, but I thought that could be an example of how we can write a code with circular invocations.

Reply:
Yes, in this particular case I agree that using the advice stack is closer to how it would be used in the real-world than hardcoding it in the code, so let's leave as-is.